client/structs/structs.go

```diff
@@ -21,6 +41,116 @@ type ClientStatsResponse struct {
 	structs.QueryMeta
 }
 
+// AllocFileInfo holds information about a file inside the AllocDir
+type AllocFileInfo struct {
+	Name     string
+	IsDir    bool
+	Size     int64
+	FileMode string
+	ModTime  time.Time
+}
+
+// FsListRequest is used to list an allocation's directory.
+type FsListRequest struct {
+	// AllocID is the allocation to list from
+	AllocID string
+
+	// Path is the path to list
+	Path string
+
+	structs.QueryOptions
+}
+
+// FsListResponse is used to return the listings of an allocation's directory.
+type FsListResponse struct {
+	// Files are the result of listing a directory.
+	Files []*AllocFileInfo
+
+	structs.QueryMeta
+}
+
+// FsStatRequest is used to stat a file
+type FsStatRequest struct {
+	// AllocID is the allocation to stat the file in
+	AllocID string
+
+	// Path is the path to list
+	Path string
+
+	structs.QueryOptions
+}
+
+// FsStatResponse is used to return the stat results of a file
+type FsStatResponse struct {
+	// Info is the result of stating a file
+	Info *AllocFileInfo
+
+	structs.QueryMeta
+}
+
+// FsStreamRequest is the initial request for streaming the content of a file.
+type FsStreamRequest struct {
+	// AllocID is the allocation to stream logs from
+	AllocID string
+
+	// Path is the path to the file to stream
+	Path string
+
+	// Offset is the offset to start streaming data at.
+	Offset int64
```

Comment:
any reason not to make this uint64, so that you can't send params like a negative offset?